cmake/find/FindAvahi.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2015 Damien Buhl
+# All rights reserved.
+
+# Try to find Avahi client and libdns_sd_compat library.
+# It can be used as :
+#  
+# find_package(Avahi COMPONENTS client compat_libdns_sd REQUIRED)
+# include_directories(AFTER ${Avahi_INCLUDE_DIRS}) 
+# 
+# target_link_libraries(program ${Avahi_LIBRARIES})
+#
+
+set(Avahi_FOUND OFF)
+set(Avahi_CLIENT_FOUND OFF)
+set(Avahi_COMPAT_LIBDNS_SD_FOUND OFF)
+
+find_library(Avahi_COMMON_LIBRARY NAMES avahi-common)
+find_library(Avahi_CLIENT_LIBRARY NAMES avahi-client)
+find_path(Avahi_CLIENT_INCLUDE_DIRS avahi-client/publish.h)
+
+if ( Avahi_COMMON_LIBRARY AND Avahi_CLIENT_LIBRARY AND Avahi_CLIENT_INCLUDE_DIRS)
+  set(Avahi_FOUND ON)
+  set(Avahi_COMMON_FOUND ON)
+  set(Avahi_CLIENT_FOUND ON)
+endif()
+
+find_library(Avahi_COMPAT_LIBDNS_SD_LIBRARY NAMES dns_sd)
+find_path(Avahi_COMPAT_LIBDNS_SD_INCLUDE_DIRS avahi-compat-libdns_sd/dns_sd.h) 
+
+if (Avahi_COMPAT_LIBDNS_SD_LIBRARY)
+  set(Avahi_COMPAT_LIBDNS_SD_FOUND ON)
+endif()
+
+set(_Avahi_MISSING_COMPONENTS "")
+foreach(COMPONENT ${Avahi_FIND_COMPONENTS})
+  string(TOUPPER ${COMPONENT} COMPONENT)
+
+  if(NOT Avahi_${COMPONENT}_FOUND)
+    string(TOLOWER ${COMPONENT} COMPONENT)
+    list(APPEND _Avahi_MISSING_COMPONENTS ${COMPONENT})
+  endif()
+endforeach()
+
+if (_Avahi_MISSING_COMPONENTS)
+  set(Avahi_FOUND OFF)
+
+  set(Avahi_ERROR_REASON
+      "${Avahi_ERROR_REASON}\nCould not find the following")
+  set(Avahi_ERROR_REASON
+    "${Avahi_ERROR_REASON} Avahi libraries:\n")
+  foreach(COMPONENT ${_Avahi_MISSING_COMPONENTS})
+    set(Avahi_ERROR_REASON
+      "${Avahi_ERROR_REASON}        ${COMPONENT}\n")
+  endforeach()
+
+  if (Avahi_FIND_REQUIRED)
+    message(SEND_ERROR "Unable to find the requested Avahi libraries.\n${Avahi_ERROR_REASON}")
+  else()
+    message(STATUS "Unable to find the requested but not required Avahi libraries.\n${Avahi_ERROR_REASON}")
+  endif()
+
+endif()
+
+
+if(Avahi_FOUND)
+  set(Avahi_LIBRARIES ${Avahi_COMMON_LIBRARY} ${Avahi_CLIENT_LIBRARY} ${Avahi_COMPAT_LIBDNS_SD_LIBRARY})
+  set(Avahi_INCLUDE_DIRS ${Avahi_CLIENT_INCLUDE_DIRS} ${Avahi_COMPAT_LIBDNS_SD_INCLUDE_DIRS})
```

Comment:
You can create modern-style imported targets instead of old-style variables with full paths to make this libs cacheable and relocatable (though they need to be relocatable by design, also all dependencies should be cacheable too). See [this overview](https://github.com/ruslo/hunter/wiki/example.find_package.config#relocation-challenge) and [boost example](https://github.com/ruslo/hunter/blob/master/cmake/templates/BoostConfig.cmake.in). But note that it's not an issue, it's an optimization note. So you can update this part later.